packages-ts/gauntlet-terra-contracts/src/commands/abstract/batchWrapper.ts

```diff
@@ -0,0 +1,109 @@
+import AbstractCommand, { makeAbstractCommand } from '.'
+import { Result, WriteCommand } from '@chainlink/gauntlet-core'
+import { TerraCommand, TransactionResponse, logger } from '@chainlink/gauntlet-terra'
+import { AccAddress, LCDClient, Msg, MsgExecuteContract, MsgSend } from '@terra-money/terra.js'
+import { prompt } from '@chainlink/gauntlet-core/dist/utils'
+import { ExecutionContext } from './executionWrapper'
+import { getRDD, parseJSON } from '@chainlink/gauntlet-terra/dist/lib/rdd'
+
+const defaultBeforeExecute = async (id: string, contracts: string[], params) => {
+  logger.loading(`Executing ${id} for the following sets of inputs`)
+  logger.log('Input Params:', params)
+  await prompt(`Continue?`)
+}
+
+export const wrapCommand = (command) => {
+  return class BatchCommand extends TerraCommand {
+    static id = `${command.id}:batch`
+    subCommands: TerraCommand[]
+
+    constructor(flags, args) {
+      super(flags, args)
+    }
+
+    validateInput = (input, flags, args): Boolean => {
+      if (input == null) {
+        if (!flags.rdd) throw new Error(`One of --input, --inputFile, or --rdd must be provided`)
+        return true
```

Comment:
```suggestion
      if (!input && !flags.rdd) throw new Error(`One of --input, --inputFile, or --rdd must be provided`)
```

Reply:
ah shoot just remembered why this change doesn't work - in the case where input is null and rdd is provided, we don't want to do any additional checks on the number of args